### ISSUE TITLE
RCS1220 produces broken suggestion when pattern matching on System.Enum

### ISSUE THREAD

**Issue by NoahStolk** [JosefPihrt/Roslynator#942]
**Product and Version Used**:

Roslynator.Analyzers 4.1.1

**Steps to Reproduce**:

Consider this method:
```csharp
Enum? Example(object? value)
{
	if (value is Enum)
		return (Enum)value;

	return null;
}
```

`RCS1220 "Use pattern matching instead of combination of 'is' operator and cast operator."` is triggered, which is correct. However, it produces a broken suggestion:

```csharp
if (value is Enum enum) // Does not compile
	return enum;
```

`enum` is a keyword, so this code does not compile. The same happens with other keywords, for example if you were to create a class named `Static`, it produces this code:

```csharp
if (value is Static static) // Does not compile
	return static;
```

`string` (or `String`) seems to work correctly:

```csharp
if (value is string x) // OK
	return x;
```

**Actual Behavior**:
```csharp
if (value is Enum enum)
	return enum;
```

**Expected Behavior**:
```csharp
if (value is Enum @enum)
	return @enum;
```

or

```csharp
if (value is Enum x)
	return x;
```
